### PR TITLE
Add serial and console schemas

### LINKED ISSUE #102: Add serial and console schemas
VM spec needs "serials" and "consoles" as below to set up a serial console in a guest virtual machine. But schema.go does not have "serials" and "consoles" schemas.

Without a serial console, "cannot find character device <null>" error occurs when attempting to connect to a guest virtual machine.

```
apiVersion: kubevirt.io/v1alpha1
kind: VM
metadata:
  name: testvm
spec:
  domain:
    devices:
      graphics:
      - autoPort: "yes"
        type: vnc
        listen:
          type: address
          address: 127.0.0.1
      interfaces:
      - source:
          network: default
        type: network
      video:
      - Model:
          heads: 1
          type: cirrus
          vram: 16384
      disks:
      - type: file
        shapshot: internal
        device: disk
        driver:
          name: qemu
          type: qcow2
        diskSource:
          file: /var/lib/libvirt/images/cirros-0.3.4-x86_64-disk.img
        diskTarget:
          dev: hda
          bus: ide
      serials:
      - type: pty
        target:
          port: 0
      consoles:
      - type: pty
        target:
          type: serial
          port: 0
    memory:
      unit: KiB
      value: 524288
    name: testvm
    os:
      type:
        os: hvm
    type: qemu
  nodeSelector:
    kubernetes.io/hostname: master
```

### PR DESCRIPTION
**Pull request by sungwonh**
VM spec needs "serials" and "consoles" elements to set up
a serial console in a guest VM.

Resolves #102

### CODE REVIEW

**Comment by rmohr** [comment]
@sungwonh the fields you added are also part of the swagger doc, that is what travis complains about.

Could you run

```
make generate
```
and commit the generated changes?

Note that the generators we are using [need to be installed](https://github.com/kubevirt/kubevirt/blob/master/docs/getting-started.md#code-generation)

File: pkg/virt-handler/virtwrap/schema.go [hunk 1]
```diff
@@ -163,6 +169,33 @@ type DiskSourceHost struct {
 
 // END Disk -----------------------------
 
+// BEGIN Serial -----------------------------
+
+type Serial struct {
+	Type         string       `xml:"type,attr"`
+	SerialTarget SerialTarget `xml:"serialTarget"`
```
Comment (rmohr) [inline pkg/virt-handler/virtwrap/schema.go]: I think the xml tag needs to be `xml:"target"`.

Also consider renaming the field name to`Target`.

File: pkg/api/v1/schema.go [hunk 2]
```diff
@@ -81,6 +83,33 @@ type DiskSourceHost struct {
 
 // END Disk -----------------------------
 
+// BEGIN Serial -----------------------------
+
+type Serial struct {
+	Type         string       `json:"type"`
+	SerialTarget SerialTarget `json:"serialTarget"`
```
Comment (rmohr) [inline pkg/api/v1/schema.go]: Naming int `Target` in the struct and `target` in the json tag should be good enough.

File: pkg/api/v1/schema.go [hunk 3]
```diff
@@ -81,6 +83,33 @@ type DiskSourceHost struct {
...
+	Port string `json:"port"`
+}
+
+// END Serial -----------------------------
+
+// BEGIN Console -----------------------------
+
+type Console struct {
+	Type          string        `json:"type"`
+	ConsoleTarget ConsoleTarget `json:"consoleTarget"`
```
Comment (rmohr) [inline pkg/api/v1/schema.go]: Naming int `Target` in the struct and `target` in the json tag should be good enough.

File: pkg/virt-handler/virtwrap/schema.go [hunk 4]
```diff
@@ -163,6 +169,33 @@ type DiskSourceHost struct {
...
+	Port string `xml:"port,attr"`
+}
+
+// END Serial -----------------------------
+
+// BEGIN Console -----------------------------
+
+type Console struct {
+	Type          string        `xml:"type,attr"`
+	ConsoleTarget ConsoleTarget `xml:"consoleTarget"`
```
Comment (rmohr) [inline pkg/virt-handler/virtwrap/schema.go]: I think the xml tag needs to be `xml:"target"`.

Also consider renaming the field name to`Target`.

File: pkg/api/v1/schema.go [hunk 5]
```diff
@@ -81,6 +83,33 @@ type DiskSourceHost struct {
...
+}
+
+type ConsoleTarget struct {
+	Type string `json:"type"`
+	Port string `json:"port"`
```
Comment (rmohr) [inline pkg/api/v1/schema.go]: I think port is optional when looking into the domain xml spec.

Could you make it `omitempty`, `uint` and a pointer?

File: pkg/virt-handler/virtwrap/schema.go (hunk 1 above)
Comment (sungwonh) [inline pkg/virt-handler/virtwrap/schema.go]: Renamed SerialTarget to Target.

File: pkg/virt-handler/virtwrap/schema.go (hunk 4 above)
Comment (sungwonh) [inline pkg/virt-handler/virtwrap/schema.go]: Renamed ConsoleTarget to Target.

File: pkg/api/v1/schema.go (hunk 2 above)
Comment (sungwonh) [inline pkg/api/v1/schema.go]: Renamed SerialTarget to Target.

File: pkg/api/v1/schema.go (hunk 3 above)
Comment (sungwonh) [inline pkg/api/v1/schema.go]: Renamed ConsoleTarget to Target.

File: pkg/api/v1/schema.go (hunk 5 above)
Comment (sungwonh) [inline pkg/api/v1/schema.go]: Changed Port type to *uint and added omitempty.

**Comment by rmohr** [comment]
ready to test

**Comment by rmohr** [comment]
ok to test

**Comment by rmohr** [comment]
@sungwonh thanks again, we will start using them right away. :cake:

**Comment by sungwonh** [comment]
@rmohr you're welcome!

**Pull request merged**